drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/wizard/column/plugins/BRLActionColumnPlugin.java

```diff
@@ -0,0 +1,301 @@
+import org.drools.workbench.screens.guided.dtable.client.wizard.column.pages.AdditionalInfoPage;
+import org.drools.workbench.screens.guided.dtable.client.wizard.column.pages.RuleModellerPage;
+import org.drools.workbench.screens.guided.dtable.client.wizard.column.plugins.commons.AdditionalInfoPageInitializer;
+import org.drools.workbench.screens.guided.dtable.client.wizard.column.plugins.commons.BaseDecisionTableColumnPlugin;
+import org.drools.workbench.screens.guided.rule.client.editor.RuleModellerConfiguration;
+import org.drools.workbench.screens.guided.rule.client.editor.events.TemplateVariablesChangedEvent;
+import org.uberfire.ext.widgets.core.client.wizards.WizardPage;
+
+import static org.drools.workbench.screens.guided.dtable.client.wizard.column.pages.common.DecisionTableColumnViewUtils.nil;
+
+@Dependent
+public class BRLActionColumnPlugin extends BaseDecisionTableColumnPlugin implements HasRuleModellerPage,
+                                                                                    HasAdditionalInfoPage,
+                                                                                    TemplateVariablesChangedEvent.Handler {
+
+    @Inject
+    private RuleModellerPage ruleModellerPage;
+
+    @Inject
+    private AdditionalInfoPage<BRLActionColumnPlugin> additionalInfoPage;
+
+    private BRLActionColumn editingCol;
+
+    private Boolean ruleModellerPageCompleted = Boolean.FALSE;
+
+    private HandlerRegistration registration;
+
+    private RuleModel ruleModel = null;
+
+    @Override
+    public String getTitle() {
+        return translate(GuidedDecisionTableErraiConstants.BRLActionColumnPlugin_AddActionBRL);
+    }
+
+    @Override
+    public List<WizardPage> getPages() {
+        return new ArrayList<WizardPage>() {{
+            add(ruleModellerPage);
+            add(additionalInfoPage());
+        }};
+    }
+
+    @Override
+    public void init(final NewGuidedDecisionTableColumnWizard wizard) {
+        super.init(wizard);
+
+        setupEditingCol();
+        setupRuleModellerEvents();
+    }
+
+    void setupRuleModellerEvents() {
+        registration = getPresenter().getEventBus().addHandler(TemplateVariablesChangedEvent.TYPE,
+                                                               this);
+    }
+
+    void setupEditingCol() {
+        editingCol = newBRLActionColumn();
+    }
+
+    void teardownRuleModellerEvents() {
+        registration.removeHandler();
+    }
+
+    @Override
+    public void onClose() {
+        super.onClose();
+
+        teardownRuleModellerEvents();
+    }
+
+    @Override
+    public Boolean generateColumn() {
+        if (nil(editingCol().getHeader())) {
+            Window.alert(translate(GuidedDecisionTableErraiConstants.BRLActionColumnPlugin_YouMustEnterAColumnHeaderValueDescription));
+            return false;
+        }
+
+        if (!isHeaderUnique(editingCol().getHeader())) {
```

Comment:
Did we really only check "action" columns in the old code for header uniqueness?